oak-auth-external/src/test/java/org/apache/jackrabbit/oak/spi/security/authentication/external/impl/principal/ExternalGroupPrincipalProviderTest.java

```diff
@@ -90,13 +90,13 @@ void sync(@NotNull ExternalUser externalUser) throws Exception {
     @NotNull
     Set<Principal> getExpectedGroupPrincipals(@NotNull String userId) throws Exception {
         if (syncConfig.user().getMembershipNestingDepth() == 1) {
-            return ImmutableSet.copyOf(idp.getUser(userId).getDeclaredGroups()).stream().map(externalIdentityRef -> {
+            return CollectionUtils.toSet(idp.getUser(userId).getDeclaredGroups()).stream().map(externalIdentityRef -> {
                 try {
                     return new PrincipalImpl(idp.getIdentity(externalIdentityRef).getPrincipalName());
                 } catch (ExternalIdentityException e) {
                     throw new RuntimeException(e);
                 }
-            }).collect(Collectors.toSet());
+            }).collect(Collectors.toUnmodifiableSet());
         } else {
```

Comment:
There seem to be two conversions to Set here (inherited from earlier).

Reply:
fixed in https://github.com/apache/jackrabbit-oak/pull/1914/commits/412cad0bd43ffc420df2ecde81ddbfadebeaeade